practice.rb

```diff
@@ -1,10 +1,36 @@
 def power(base,exp)
+	return 1 if exp==0 
+	return base if exp==1
+	return base*power(base,exp-1)
 end
 
 def fib(n)
+	return n if n==0 or n==1
+	return fib(n-1)+fib(n-2)
 end
 
 def pal(s)
+	return true if s=="" 
+	length=s.length
+	midpont=s.length/2
+	
+	#delete middle character if string has odd number of letters
+	s[midpont]='' if length%2 != 0
+	
+	#if only two letters left, check if they are the same.
+	#if they are then it is a palidrome (return true)
+	#otherwise it is not a palidrome
+	if length==2 
+		if s[0]==s[1]
+			return true
+		else
+			return false
+		end
+	end
+
+	#see if the two middle letters are the same 
+	#and if they are check if the rest of the middle letters are the same
+	return (pal(s.slice!(midpont-1..midpont)) and pal(s))
 end
```

Comment:
Interesting. Looks pretty good but could have been simplified if you added a base case `return true if s.length == 1"

I also don't think you need the `and pal(s)` at the end......